src/sensors/distant.cpp

```diff
@@ -73,12 +172,26 @@ MTS_VARIANT class DistantSensor final : public Sensor<Float, Spectrum> {
                     ScalarPoint3f(0.0f), ScalarPoint3f(direction), up));
         }
 
-        if (props.has_property("target")) {
-            m_target     = props.point3f("target");
-            m_has_target = true;
-            Log(Debug, "Targeting point %s", m_target);
-        } else {
-            m_has_target = false;
+        if constexpr (OriginType == RayOriginType::Disk) {
+            if (!props.has_property("origin_center") || !props.has_property("origin_radius")) {
+                Throw("Circular origin requires the 'origin_center' and 'origin_radius' parameters");
```

Comment:
Please use property keys or enum values in your error messages for better error reporting.

Reply:
Is it better now?